pkg/cloud/services/networking/securitygroups.go

```diff
@@ -393,6 +393,9 @@ func (s *Service) generateDesiredSecGroups(openStackCluster *infrav1.OpenStackCl
 func (s *Service) GetSecurityGroups(securityGroupParams []infrav1.SecurityGroupParam) ([]string, error) {
 	var sgIDs []string
 	for _, sg := range securityGroupParams {
+		if sg.UUID == "" {
+			continue
+		}
```

Comment:
What's this doing? Doesn't this make it impossible to use the filter functionality?

Reply:
This looks strange I agree. Will investigate.

Reply:
This was definitely a mistake. Perhaps something left from debugging.